File: src/main/java/com/gandan/practice/task/TaskReflection.java
package com.gandan.practice.task;

import com.gandan.practice.reflection.SampleClassReflection;

public class TaskReflection implements ITask {
    @Override
    public void execute() {
        SampleClassReflection sample = new SampleClassReflection(String.class);
        sample.displayClassnameAtRuntime();
        sample.dipslayModifier();

        System.out.print("Generate Empty String with newInstance");
        String emptyString = (String)sample.newInstance();
        System.out.println(" " + "".equals(emptyString));
    }
}

File: src/main/kotlin/com/gandan/practice/task/TaskReflection.kt
package com.gandan.practice.task

import com.gandan.practice.reflection.SampleClassReflection

class TaskReflection : ITask {
    override fun execute() {
        val sample = SampleClassReflection(String::class.java)
        sample.displayClassnameAtRuntime()
        sample.dipslayModifier()

        print("Generate Empty String with newInstance")
        val emptyString = sample.newInstance() as String
        println(" " + ("" == emptyString))
    }
}
